perf_test/sparse/KokkosSparse_run_spgemm.hpp

```diff
@@ -25,82 +27,93 @@ namespace KokkosKernels {
 
 namespace Experiment {
 template <typename crsMat_t, typename device>
-bool is_same_matrix(crsMat_t output_mat1, crsMat_t output_mat2) {
+bool is_same_matrix(crsMat_t output_mat_actual, crsMat_t output_mat_reference) {
   typedef typename crsMat_t::StaticCrsGraphType graph_t;
   typedef typename graph_t::row_map_type::non_const_type lno_view_t;
   typedef typename graph_t::entries_type::non_const_type lno_nnz_view_t;
   typedef typename crsMat_t::values_type::non_const_type scalar_view_t;
 
-  size_t nrows1    = output_mat1.graph.row_map.extent(0);
-  size_t nentries1 = output_mat1.graph.entries.extent(0);
-  size_t nvals1    = output_mat1.values.extent(0);
+  size_t nrows1    = output_mat_actual.graph.row_map.extent(0);
+  size_t ncols1    = output_mat_actual.graph.row_map.extent(0);
+  size_t nentries1 = output_mat_actual.graph.entries.extent(0);
+  size_t nvals1    = output_mat_actual.values.extent(0);
 
-  size_t nrows2    = output_mat2.graph.row_map.extent(0);
-  size_t nentries2 = output_mat2.graph.entries.extent(0);
-  size_t nvals2    = output_mat2.values.extent(0);
+  size_t nrows2    = output_mat_reference.graph.row_map.extent(0);
+  size_t ncols2    = output_mat_reference.graph.row_map.extent(0);
+  size_t nentries2 = output_mat_reference.graph.entries.extent(0);
+  size_t nvals2    = output_mat_reference.values.extent(0);
 
-  KokkosSparse::sort_crs_matrix(output_mat1);
-
-  if (nrows1 != nrows2) {
-    std::cerr << "row count is different" << std::endl;
+  if (nrows1 != nrows2 || ncols1 != ncols2) {
```

Comment:
Seems redundant to check rows and cols when both variables are set as `graph.row_map.extent(0)` which actually means we assume matrices to be square.

Reply:
Both the Ccrsmat_result and Ccrsmat_ref crsmats are constructed to be m*k, so yeah it's pretty much redundant but I wouldn't mind keeping it as a sanity check for if this test changes in the future

Reply:
Sure, I'm not against it either.
Let's see how the auto-tester does, I can re-approve then